Update `wrangler types` documentation with new customization options (#13040)
cloudflare/cloudflare-docs: content/workers/wrangler/commands.md

Generate types from bindings and module rules in configuration.

```sh
wrangler types [<PATH>] [OPTIONS]
```

{{<definitions>}}

- `PATH` {{<type>}}string{{</type>}} {{<prop-meta>}}(default: `worker-configuration.d.ts`){{</prop-meta>}}

>   - The path to where to write the resulting types n entry point for your Worker.

Reviewer: "n"?

Reviewer: Perhaps:

```suggestion
  - The path to where the declaration file for your Worker will be written.
```

Author: > "n"?

sorry... I have no idea what happened there 😓

>   - Since it is a declaration file, it needs to have a `d.ts` extension.

Reviewer 2: ```suggestion
  - A declaration file needs to have a `d.ts` extension.
```

Reviewer: I don't think that suggestion quite captures the validation here. What we are trying to say is that this option is invalid and will error if it does not end in `.d.ts`.

Perhaps:

```suggestion
  - The path to the declaration file must have a `d.ts` extension.
```

Author: Thanks @petebacondarwin that looks good to me 😄👍 

Hopefully @deadlypants1973 agrees 🙂

- `--env-interface` {{<type>}}string{{</type>}} {{<prop-meta>}}(default: `Env`){{</prop-meta>}}
  - The name of the interface to generate for the environment object.

>   - _Not valid if the worker uses the Service Worker syntax._

Reviewer 2: ```suggestion
  - Not valid if the Worker uses the Service Worker syntax.
```

>   - _Ignored if the worker uses the Service Worker syntax._

Reviewer: Not valid if the worker uses Service Worker syntax.

Author: 🤦 yeah I need to update this thanks! 🙇

Author: Updated 🙂👍
